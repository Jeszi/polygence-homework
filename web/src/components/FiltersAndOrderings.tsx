import React from "react";

import {
  FiltersWrapper,
  Orderings,
  CurrencyFilters,
  CurrencyButton,
} from "../styles/ComponentStyles";

interface FiltersAndOrderingsProps {
  setCurrencyFilter: (currency: string) => void;
  currencyFilter: string;
  setSorting: (value: string) => void;
}

const currencyFilters = ["ALL", "USD", "HUF"];

export const FiltersAndOrderings: React.FC<FiltersAndOrderingsProps> = (
  props
) => {
  function handleFilter(currency: string) {
    props.setCurrencyFilter(currency);
  }

  function handleSort(event: React.ChangeEvent<HTMLSelectElement>) {
    event.preventDefault();
    props.setSorting(event.target.value);
  }

  return (
    <>
      <FiltersWrapper>
        <Orderings>
          <select onChange={handleSort}>
            <option value="spent_at|desc">
              Sort by Date descending (default)
            </option>
            <option value="spent_at|asc">Sort by Date ascending</option>
            <option value="amount|desc">Sort by Amount descending</option>
            <option value="amount|asc">Sort by Amount ascending</option>
          </select>
        </Orderings>
        <CurrencyFilters>
          {currencyFilters.map((currencyFilter) => (
            <li key={currencyFilter}>
              <CurrencyButton
                data-testid={`filter-button-${currencyFilter}`}
                onClick={() => handleFilter(currencyFilter)}
                name={currencyFilter}
                currencyFilter={props.currencyFilter}
              >
                {currencyFilter}
              </CurrencyButton>
            </li>
          ))}
        </CurrencyFilters>
      </FiltersWrapper>
    </>
  );
};
