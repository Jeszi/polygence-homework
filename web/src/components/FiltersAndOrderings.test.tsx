import React from "react";
import { renderWithProviders } from "../testing/test-utils";

import { FiltersAndOrderings } from "./FiltersAndOrderings";

describe("FiltersAndOrderings component", () => {
  const props = {
    setCurrencyFilter: jest.fn(),
    currencyFilter: "ALL",
    setSorting: jest.fn(),
  };

  it("should render component", async () => {
    const { asFragment } = renderWithProviders(
      <FiltersAndOrderings {...props} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
