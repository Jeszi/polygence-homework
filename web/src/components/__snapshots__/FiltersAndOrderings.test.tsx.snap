// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersAndOrderings component should render component 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL dQIffU"
  >
    <div
      class="sc-hKwDye fwPSRI"
    >
      <select>
        <option
          value="spent_at|desc"
        >
          Sort by Date descending (default)
        </option>
        <option
          value="spent_at|asc"
        >
          Sort by Date ascending
        </option>
        <option
          value="amount|desc"
        >
          Sort by Amount descending
        </option>
        <option
          value="amount|asc"
        >
          Sort by Amount ascending
        </option>
      </select>
    </div>
    <ul
      class="sc-gsDKAQ fWdgni"
    >
      <li>
        <button
          class="sc-dkPtRN cPwnpl"
          data-testid="filter-button-ALL"
          name="ALL"
        >
          ALL
        </button>
      </li>
      <li>
        <button
          class="sc-dkPtRN bNGCDu"
          data-testid="filter-button-USD"
          name="USD"
        >
          USD
        </button>
      </li>
      <li>
        <button
          class="sc-dkPtRN bNGCDu"
          data-testid="filter-button-HUF"
          name="HUF"
        >
          HUF
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
